packages/ui/src/components/editor/input/Euler/index.tsx

```diff
@@ -43,66 +43,71 @@ type EulerInputProps = {
   unit?: string
 }
 
+const temp_Euler = new Euler()
+const temp_Quat = new Quaternion()
+
 /**
  * FileIEulerInputnput used to show EulerInput.
  *
  * @type {Object}
  */
 export const EulerInput = (props: EulerInputProps) => {
   const quaternion = useHookstate(props.quaternion)
-  const euler = useHookstate(new Euler().setFromQuaternion(quaternion.value, 'YXZ'))
+  const euler = useHookstate(temp_Euler.setFromQuaternion(props.quaternion, 'YXZ'))
 
   useEffect(() => {
-    euler.value.setFromQuaternion(quaternion.value, 'YXZ')
+    euler.set(temp_Euler.setFromQuaternion(quaternion.value, 'YXZ'))
```

Comment:
`tempEuler` persists the changes. We can use `new Euler().<methodName>` without needing tempEuler (or tempQuartenion)

```suggestion
    euler.set(new Euler().setFromQuaternion(quaternion.value, 'YXZ'))
```

Reply:
using a temp euler to not have to create a new one every time EulerInput gets rerendered

Reply:
tempEuler will be overritten before the euler value gets set